src/equations/compressible_euler_multicomponent_1d.jl

```diff
@@ -202,16 +202,15 @@ function initial_condition_weak_blast_wave(x, t,
     r = abs(x_norm)
     cos_phi = x_norm > 0 ? 1 : -1
 
-    prim_rho = SVector{ncomponents(equations), real(equations)}(r > 0.5f0 ?
-                                                                2^(i - 1) * (1 - 2) /
-                                                                (1 -
-                                                                 2^ncomponents(equations)) *
-                                                                one(RealT) :
-                                                                2^(i - 1) * (1 - 2) /
-                                                                (1 -
-                                                                 2^ncomponents(equations)) *
-                                                                convert(RealT, 1.1691)
-                                                                for i in eachcomponent(equations))
+    prim_rho = SVector{ncomponents(equations), RealT}(r > 0.5f0 ?
+                                                      2^(i - 1) * (1 - 2) /
+                                                      (1 -
+                                                       2^ncomponents(equations)) :
+                                                      2^(i - 1) * (1 - 2) /
```

Comment:
This version also works and it has no `Float64` if single precision is chosen. 
```julia
if RealT == Float32
        prim_rho = SVector{ncomponents(equations), RealT}(r > 0.5f0 ?
                                                          2^(i - 1) * (1 - 2) * 1.0f0 /
                                                          (1 -
                                                           2^ncomponents(equations)) :
                                                          2^(i - 1) * (1 - 2) /
                                                          (1 -
                                                           2^ncomponents(equations)) *
                                                          Float32(1.1691)
                                                          for i in eachcomponent(equations))
    end

    if RealT == Float64
        prim_rho = SVector{ncomponents(equations), RealT}(r > 0.5 ?
                                                          2^(i - 1) * (1 - 2) * 1.0 /
                                                          (1 -
                                                           2^ncomponents(equations)) :
                                                          2^(i - 1) * (1 - 2) /
                                                          (1 -
                                                           2^ncomponents(equations)) *
                                                          1.1691
                                                          for i in eachcomponent(equations))
    end
```
But it looks bad - which one do you prefer?